core/eolearn/tests/test_core_tasks.py

```diff
@@ -199,46 +199,30 @@ def test_remove_fails(patch: EOPatch) -> None:
         RemoveFeatureTask((FeatureType.BBOX, None))(patch)
 
 
-def test_duplicate_feature(patch):
-    mask_data = np.arange(10).reshape(5, 2, 1, 1)
-    feature_name = "MASK1"
-    duplicate_name = "MASK2"
-
-    patch = AddFeatureTask((FeatureType.MASK, feature_name))(patch, mask_data)
+@pytest.mark.parametrize(
+    "feature_specification",
+    [
+        [(FeatureType.DATA, "bands", "bands2")],
+        [(FeatureType.DATA, "bands", "bands2"), (FeatureType.MASK_TIMELESS, "mask", "mask2")],
+        [(FeatureType.DATA, "bands", f"bands{i}") for i in range(5)],
+    ],
+)
+@pytest.mark.parametrize("deep", [True, False])
+def test_duplicate_feature(feature_specification: List[FeaturesSpecification], deep: bool, patch: EOPatch) -> None:
+    patch = DuplicateFeatureTask(feature_specification, deep)(patch)
 
-    duplicate_task = DuplicateFeatureTask((FeatureType.MASK, feature_name, duplicate_name))
-    patch = duplicate_task(patch)
+    for f_type, f_name, f_dup_name in feature_specification:
+        assert f_dup_name in patch[f_type]
+        original_id = id(patch[(f_type, f_name)])
+        duplicated_id = id(patch[(f_type, f_dup_name)])
+        assert original_id != duplicated_id if deep else original_id == duplicated_id
+        assert_array_equal(patch[(f_type, f_name)], patch[(f_type, f_dup_name)])
```

Comment:
just another 'prettification' suggestion.
I would do `orig_feat = f_type, f_name` and `dup_feat = f_type, f_dup_name` (or use longer names `original_feature` if it makes the code prettier).
You can then do `assert duplicated_feature in patch` and you get rid of a lot of parenthesis later on (and it is more clear whether you're looking for the original or duplicate)

Reply:
done